### ISSUE TITLE
AttributeError: 'Traceback' object has no attribute 'tb_lasti'

### ISSUE THREAD

**Issue by gavinwahl** [getsentry/raven-python#740]
I'm getting this error when using raven with celery:

```
Traceback (most recent call last):
  File "raven/handlers/logging.py", line 66, in emit
    return self._emit(record)
  File "raven/contrib/django/handlers.py", line 33, in _emit
    return super(SentryHandler, self)._emit(record, request=request)
  File "raven/handlers/logging.py", line 184, in _emit
    extra=extra, date=date, **kwargs)
  File "raven/contrib/django/client.py", line 167, in capture
    result = super(DjangoClient, self).capture(event_type, **kwargs)
  File "raven/base.py", line 543, in capture
    if self.skip_error_for_logging(exc_info):
  File "raven/base.py", line 286, in skip_error_for_logging
    key = self._get_exception_key(exc_info)
  File "raven/base.py", line 278, in _get_exception_key
    exc_info[0],
AttributeError: 'Traceback' object has no attribute 'tb_lasti'
```

It seems like the issue is that exc_info[2] is a billiard.einfo.Traceback object, not a regular python traceback. The Traceback object in billiard does not have a tb_lasti property, but python's does.

**Comment by mitsuhiko** [comment]
Fixed in 865b3f9ab459da2bf7e3ba3fce3293ec20346d45

**Comment by gavinwahl** [comment]
There is a traceback, it just doesn't have the tb_lasti property. So this still raises the same error.

**Comment by hheimbuerger** [comment]
@mitsuhiko A cursory review of 865b3f9 lets me agree with @gavinwahl. Can you please either reopen the ticket or explain how this commit fixes the issue?

**Comment by hheimbuerger** [comment]
At any rate, seems like it has been fixed in [b945d6b](https://github.com/celery/billiard/commit/b945d6b392e1fa44a43c0e130e675f0a3aab4f84) on celery (actually: billiard) for https://github.com/celery/billiard/issues/180.

So I guess this shouldn't be a problem anymore for the mentioned case (Raven + Celery) and can stay closed.

**Comment by mitsuhiko** [comment]
@gavinwahl indeed, I just noticed that this was a slightly different error. I don't know if i should deal with bad traceback objects. This is clearly a bug in billiard that apparently got closed.  I will consider it though.  Need to check what billiard's patch status is.

**Comment by hheimbuerger** [comment]
@mitsuhiko From what I read (again, haven't tested it), this should have been released with [billiard 3.3.0.23](https://pypi.python.org/pypi/billiard/3.3.0.23), which came out this morning (even though there's no release on GitHub), and is depended on by celery 3.1.23.
